server/src/main/java/io/crate/planner/node/management/ExplainPlan.java

```diff
@@ -208,6 +196,172 @@ public void executeOrFail(DependencyCarrier dependencies,
+
+        Timer timer = context.createTimer(Phase.Execute.name());
+        timer.start();
+        BaseResultReceiver resultReceiver = new BaseResultReceiver();
+        RowConsumer noopRowConsumer = new RowConsumerToResultReceiver(resultReceiver, 0, t -> {});
+        NodeOperationTree operationTree = LogicalPlanner.getNodeOperationTree(
+            plan, executor, plannerContext, params, subQueryResults);
+
+        resultReceiver.completionFuture()
+            .whenComplete(createResultConsumer(executor, consumer, plannerContext.jobId(), timer, operationTree, subQueryExplainResults));
+
+        LogicalPlanner.executeNodeOpTree(
+            executor,
+            plannerContext.transactionContext(),
+            plannerContext.jobId(),
+            noopRowConsumer,
+            true,
+            operationTree
+        );
+        return consumer.completionFuture();
+    }
+
+    private CompletableFuture<?> executePlan(LogicalPlan plan,
+                                             DependencyCarrier executor,
+                                             PlannerContext plannerContext,
+                                             RowConsumer consumer,
+                                             Row params,
+                                             SubQueryResults subQueryResults,
+                                             Map<SelectSymbol, Object> valuesBySubQuery,
+                                             List<Map<String, Object>> explainResults,
+                                             @Nullable RamAccounting ramAccounting,
+                                             @Nullable SelectSymbol selectSymbol) {
+        boolean isTopLevel = selectSymbol == null;
+
+        assert ramAccounting != null || isTopLevel : "ramAccounting must NOT be null for subPlans";
+
+        if (ramAccounting == null) {
+            ramAccounting = ConcurrentRamAccounting.forCircuitBreaker(
+                "multi-phase",
+                executor.circuitBreaker(HierarchyCircuitBreakerService.QUERY),
+                plannerContext.transactionContext().sessionSettings().memoryLimitInBytes()
+            );
+        }
+
+        if (isTopLevel == false) {
+            plannerContext = PlannerContext.forSubPlan(plannerContext);
+        }
+
+        IdentityHashMap<SelectSymbol, Object> subPlanValueBySubQuery = new IdentityHashMap<>();
+        ArrayList<Map<String, Object>> subPlansExplainResults = new ArrayList<>(plan.dependencies().size());
+        List<CompletableFuture<?>> subPlansFutures = new ArrayList<>(plan.dependencies().size());
+
+        for (Map.Entry<LogicalPlan, SelectSymbol> entry : plan.dependencies().entrySet()) {
+            SelectSymbol subPlanSelectSymbol = entry.getValue();
+            // TODO: do we need to optimize the subPlan here? Isn't it already optimized?
+            final LogicalPlan subPlan = plannerContext.optimize().apply(entry.getKey(), plannerContext);
```

Comment:
Isn't the sub-plan also already optimized?

Reply:
From a quick try, it seems that it's not needed, but of course you can check it further and remove it in a separate PR.

Reply:
yeah will followup here.

Reply:
If I remember right it came up in https://github.com/crate/crate/pull/15840 
The optimizer rule can add new subqueries and instead of re-optimizing after another rule we deferred the optimizations to just before the plans are used.

Reply:
Ah ok, thanks for pointing out! I'll remove the todo then and replace it with a comment.